### ISSUE TITLE
Question about translate directive with input element value

### ISSUE THREAD

**Issue by EslaMx7** [robisim74/angular-l10n#48]
I wonder if there is any workaround to make the translate directive works in this HTML template:

```HTML
<div class="btn-container">
      <input type="submit" class="btn1" value="Login" translate />
</div>
```

This actually doesn't work without using the `TranslatePipe` like this:

```HTML
<div class="btn-container">
      <input type="submit" class="btn1" value="{{'Login' | translate}}" />
</div>
```

This works great, **but** I want to keep my HTML clean because another person is working on the UI separately and we agreed to some rules between us regarding the localization to use the translate directive when possible.

I made a workaround by using `<button type="submit">` element instead of `<input>`, **And it worked**.
 **However** this will not be acceptable if the CSS is applied to a specific element type like `.input[type="submit"]` which it is in my case.

With some investigations in the source code, I think the problem comes from [here](https://github.com/robisim74/angular2localization/blob/master/src/directives/translate.directive.ts#L46-L54):

```JavaScript
    private translate(renderNode: any, nodeValue: string, key: string): void {

        this.localization.translateAsync(key, this.params).subscribe(
            (value: string) => {
                this.renderer.setText(renderNode, nodeValue.replace(key, value));
            }
        );

    }
```

Since we cannot `setText` on an input element, I suggest checking the type of the `renderNode` before setting the text, and to support translating input values we might use 'setElementAttribute' function.

**Comment by robisim74** [comment]
@EslaMx7 Thank you very much for the carefull analysis.

If I understand correctly, whether a tag has `value` and `translate` attributes, the directive shoud use the `value` as key to translate and not the text:

`<tagname value="TITLE" translate></tagname>`

this should be rendered as:

`<tagname value="Angular 2 localization" translate></tagname>`

Ok, it's possible. My question: is this statement always true? when the text isn't there, is there always the `value` attribute?

**Comment by EslaMx7** [comment]
@robisim74 I cannot confirm that this statement is always true, but for input (form controls) elements it is.
After a little search, I got some evidences: 

> Setting the `value` is normally used for input/form elements. `innerHTML` is normally used for `div`, `span`, `td` and similar elements.
> 
> `value` applies only to objects that have the `value` attribute (**normally, form controls**).
> 
> `innerHtml` applies to every object that can contain HTML (divs, spans, but many other and also form controls).
> 
> _They are not equivalent or replaceable._

[Source](http://stackoverflow.com/questions/20604299/what-is-innerhtml-on-input-elements)

---

> **innerHTML** : This property was initially implemented by web browsers, then specified by the WHATWG and W3C in HTML5. Old implementations may not all implement it exactly the same way. For example, when text is entered into a text input, Internet Explorer changes the value attribute of the input's innerHTML property but Gecko browsers do not.

[Source](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML)

----

> **value**
> The initial value of the control. This attribute is optional except when the value of the type attribute is radio or checkbox.

[Source](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-value)

---

So, I think you may make a special case for form controls inputs to change the value attribute, I could propose a PR if you agreed to add this in the next version.

Thank you very much for your great work, I really loved working with it.

**Comment by robisim74** [comment]
I would not extend the use of directives to the attributes, but in this case you have convinced me.

If you have time to do a PR, you are welcome.

Thank you again,

Roberto

**Comment by robisim74** [comment]
I included this feature in the new major version. Greetings

**Comment by FabienDehopre** [comment]
this feature you have introduced produced some strange issues on my website. 
in a select, it does not translate the option text but only the option value.
I, personally, think that the translate directive shouldn't be used to translate input or option values.

Here is an example:
The following html template
```html
<select [(ngModel)]="itemsPerPage" (change)="pageSizeChanged()">
  <option value="10">10</option>
  <option value="20">20</option>
  <option value="50">50</option>
  <option value="100">100</option>
  <option value="-1" translate>PAGE_SIZE_ALL</option>
 </select>
```
produces the following html
```html
<select _ngcontent-cti-46="" ng-reflect-model="10" class="ng-pristine ng-valid ng-touched">
  <option _ngcontent-cti-46="" value="10" ng-reflect-value="10">10</option>
  <option _ngcontent-cti-46="" value="20" ng-reflect-value="20">20</option>
  <option _ngcontent-cti-46="" value="50" ng-reflect-value="50">50</option>
  <option _ngcontent-cti-46="" value="100" ng-reflect-value="100">100</option>
  <option _ngcontent-cti-46="" translate="" value="Key Not Found" ng-reflect-value="-1">PAGE_SIZE_ALL</option>
</select>
```
Note that the last ```<option>``` value has been translated instead of the text. This is not the desired behavior.

**Comment by robisim74** [comment]
@FabienDehopre I will try to fix it in this way: if there is a node value, it translates the node value, otherwise the _value_ attribute: this solution should work also with _checkbox_ and _radio_.
@EslaMx7 If that doesn't work in all situations, I will revert the feature.

Tonight I will release a new version (I already have another fix to do).

**Comment by robisim74** [comment]
Fixed in v2.0.2: https://github.com/robisim74/angular-l10n/releases: now _value_ attribute is translated only if there isn't a valid text in the child node.

**Comment by EslaMx7** [comment]
@FabienDehopre Thank you for this important note, it should be working as expected now.
@robisim74 That's really great, Thank you for the fix, everything is making sense now regarding the _node value_ and the _value attribute_, I hope this fits all.
